### PR TITLE
Add support for DNF (Yum v4) property `install_weak_deps`

### PR DESCRIPTION
**Pull request by newatlantis**
Signed-off-by: Lincoln Abbey <linc.abbey@gmail.com>

# Description

Describe what this change achieves

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,6 +58,18 @@ end
 - `exactarch` - Either `true` or `false`. Set to `true` to make 'yum update' only update the architectures of packages that you have installed. ie: with this enabled yum will not install an i686 package to update an x86_64 package. Default is `true`
 - `gpgcheck` - Either `true` or `false`. This tells yum whether or not it should perform a GPG signature check on the packages gotten from this repository.
 
+#### Yum v4 DNF Properties
```
[the remark this answers is not in the document]
Comment (bmhughes) [reply README.md:61]: The weak deps options is valid for yum as well so just put it under the original `#### Properties` header.

**Review by bmhughes — CHANGES_REQUESTED** [review]

**Comment by bmhughes** [comment]
Just a couple of small changes and this is good to go.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,8 +4,14 @@ This file is used to list changes made in each version of the yum cookbook.
 
 ## Unreleased
 
+## 7.1.0 - *2021-08-26*
+
+- Add support for DNF (Yum v4) property `install_weak_deps` (#193)
```
[the remark this answers is not in the document]
Comment (newatlantis) [reply CHANGELOG.md:7]: OK, I think I understand what you mean. Thank you, I didn't know that. Did that.

**Review by newatlantis — COMMENTED** [review]
Thank you for your help with this PR. I'm still getting used to the GitHub interface and capabilities. So I hope I did it right. I added some comments (more like questions) to the PR itself, hoping I could get your feedback on a few points that are not clear to me. Please let me know if you see me not following the desired process here correctly!

**Comment by newatlantis** [comment]
> Just a couple of small changes and this is good to go.

@bmhughes: I tried to address the two changes I saw you request, to the CHANGELOG.md and README.md files. It looks like the CI build has passed. But I see above the message that "Merging can be performed automatically once the requested changes are addressed." Could I trouble you to point out to me what I need to do next for this PR? Thanks for all your help.

File: test/fixtures/cookbooks/yum_test/recipes/test_globalconfig_two.rb [hunk 3]
```diff
@@ -36,6 +36,7 @@
   history_record_packages 'rpm'
   http_caching 'all'
   ip_resolve '4'
+  install_weak_deps false
```
[the remark this answers is not in the document]
Comment (bmhughes) [reply test/fixtures/cookbooks/yum_test/recipes/test_globalconfig_two.rb:39]: This is fine (initially at least)

File: templates/main.erb [hunk 4]
```diff
@@ -116,6 +116,13 @@ http_caching=<%= @config.http_caching %>
 <% if @config.ip_resolve %>
 ip_resolve=<%= @config.ip_resolve %>
 <% end %>
+<% unless @config.install_weak_deps.nil? %>
```
[the remark this answers is not in the document]
Comment (bmhughes) [reply templates/main.erb:119]: You don't need the double negative of `unless` and the `.nil?` call and the additional if/else block.

You can just do `if @config.install_weak_deps` as the other properties are and then use `coerce` on the property to set the integer value higher up the call stack. I've put what I mean as a comment against the property declaration. The default value is true so we can set that by default without issue.

File: resources/globalconfig.rb [hunk 5]
```diff
@@ -79,6 +79,7 @@
...
 property :installonlypkgs, String, description: 'List of package provides that should only ever be installed, never updated.  Kernels in particular fall into this category. Defaults to kernel, kernel-bigmem, kernel-enterprise, kernel-smp, kernel-modules, kernel-debug, kernel- unsupported, kernel-source, kernel-devel, kernel-PAE, kernel- PAE-debug.'
 property :installroot, String, description: 'Specifies an alternative installroot, relative to which all packages will be installed.'
+property :install_weak_deps, [true, false], description: "When this option is set to true and a new package is about to be installed, all packages linked by a weak dependency relation (i.e., Recommends or Supplements flags) with this package will be pulled into the transaction. Default is DNF's default of true."
```
Comment (bmhughes) [inline resources/globalconfig.rb:82]: ```property :install_weak_deps, [true, false], default: true, coerce: proc { |p| p ? 1 : 0 }, description: "When this option is set to true and a new package is about to be installed, all packages linked by a weak dependency relation (i.e., Recommends or Supplements flags) with this package will be pulled into the transaction. Default is DNF's default of true."```

**Review by bmhughes — CHANGES_REQUESTED** [review]

File: templates/main.erb (hunk 4 above)
[the remark this answers is not in the document]
Comment (bmhughes) [reply templates/main.erb:119]: We can drop all of this and just do a check against the property value being false and then set `install_weak_deps=0`, the default value is true so we don't need to worry about it until it's needed to be set to false.

**Comment by bmhughes** [comment]
> 
> 
> > Just a couple of small changes and this is good to go.
> 
> @bmhughes: I tried to address the two changes I saw you request, to the CHANGELOG.md and README.md files. It looks like the CI build has passed. But I see above the message that "Merging can be performed automatically once the requested changes are addressed." Could I trouble you to point out to me what I need to do next for this PR? Thanks for all your help.

That'll dissapear once I (or someone else) approves the PR.

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [7.1.0](https://github.com/sous-chefs/yum/releases/tag/7.1.0)

**Pull request merged**